test/mlir/onnx/onnx_constprop.mlir

```diff
@@ -379,6 +490,266 @@ func.func @test_div_ones(%arg0 : tensor<1x2xui8>) -> tensor<1x2xui8> {
+  return %0 : tensor<3x2xbf16>
+// CHECK-LABEL:  func @test_clip_max_and_min
+  // CHECK: {{.*}} = onnx.Constant dense<{{.}}[2.000000e+00, -2.000000e+00], [0.000000e+00, 0x7FC0], [-2.000000e+00, 2.000000e+00]]>
+  // CHECK-NOT: {{.*}} = "onnx.Clip"
+}
+
+// CHECK-LABEL:  func @test_clip_no_min
+func.func @test_clip_no_min() -> tensor<3x2xbf16> {
+  // Test Positive Clamped, Negative Clamped, In range, NaN, -Inf,  +Inf
+  %cst = onnx.Constant dense<[[2.1, -2.125], [0.0, 0x7FC0], [0xFF80, 0x7F80]]> : tensor<3x2xbf16>
+  %none = "onnx.NoValue"() {value} : () -> none
+  %max = onnx.Constant {value = dense<[2.0]> : tensor<1xbf16>} : tensor<1xbf16>
+  %0 = "onnx.Clip"(%cst, %none, %max) : (tensor<3x2xbf16>, none, tensor<1xbf16>) -> tensor<3x2xbf16>
+  return %0 : tensor<3x2xbf16>
+  // CHECK: {{.*}} = onnx.Constant dense<{{.}}[2.000000e+00, -2.125000e+00], [0.000000e+00, 0x7FC0], [-3.389530e+38, 2.000000e+00]]>
+  // CHECK-NOT: {{.*}} = "onnx.Clip"
+}
+
+// CHECK-LABEL:  func @test_clip_no_max
+func.func @test_clip_no_max() -> tensor<3x2xbf16> {
+  // Test Positive Clamped, Negative Clamped, In range, NaN, -Inf,  +Inf
+  %cst = onnx.Constant dense<[[2.125, -2.125], [0.0, 0x7FC0], [0xFF80, 0x7F80]]> : tensor<3x2xbf16>
+  %min = onnx.Constant {value = dense<-2.0> : tensor<bf16>} : tensor<bf16>
+  %none = "onnx.NoValue"() {value} : () -> none
+  %0 = "onnx.Clip"(%cst, %min, %none) : (tensor<3x2xbf16>, tensor<bf16>, none) -> tensor<3x2xbf16>
+  return %0 : tensor<3x2xbf16>
+  // CHECK: {{.*}} = onnx.Constant dense<{{.}}[2.125000e+00, -2.000000e+00], [0.000000e+00, 0x7FC0], [-2.000000e+00, 3.389530e+38]]>
+  // CHECK-NOT: {{.*}} = "onnx.Clip"
+}
+
+// CHECK-LABEL:  func @test_clip_multiple_min
+func.func @test_clip_multiple_min() -> tensor<3x2xbf16> {
+  // Test Positive Clamped, Negative Clamped, In range, NaN, -Inf,  +Inf
+  %cst = onnx.Constant dense<[[2.125, -2.125], [0.0, 0x7FC0], [0xFF80, 0x7F80]]> : tensor<3x2xbf16>
+  %min = onnx.Constant dense<[[5.000, 2.000], [-1.0, 0.0], [0.0, 0.125]]> : tensor<3x2xbf16>
+  %max = onnx.Constant {value = dense<[10.0]> : tensor<1xbf16>} : tensor<1xbf16>
+  %0 = "onnx.Clip"(%cst, %min, %max) : (tensor<3x2xbf16>, tensor<3x2xbf16>, tensor<1xbf16>) -> tensor<3x2xbf16>
+  return %0 : tensor<3x2xbf16>
+  // CHECK: {{.*}} = onnx.Constant dense<{{.}}[5.000000e+00, 2.000000e+00], [0.000000e+00, 0x7FC0], [0.000000e+00, 1.000000e+01]]>
+  // CHECK-NOT: {{.*}} = "onnx.Clip"
+}
+
+// CHECK-LABEL:  func @test_clip_multiple_max
+func.func @test_clip_multiple_max() -> tensor<3x2xbf16> {
+  // Test Positive Clamped, Negative Clamped, In range, NaN, -Inf,  +Inf
+  %cst = onnx.Constant dense<[[2.125, -2.125], [0.0, 0x7FC0], [0xFF80, 0x7F80]]> : tensor<3x2xbf16>
+  %min = onnx.Constant {value = dense<-2.0> : tensor<bf16>} : tensor<bf16>
+  %max = onnx.Constant dense<[[5.000, 2.000], [-1.0, 0.0], [0.0, 0.125]]> : tensor<3x2xbf16>
```

Comment:
`min` and `max` must be a scalar according to its specs: https://onnx.ai/onnx/operators/onnx__Clip.html

It would fail once we verify this in the ClipOp's verifier.

Reply:
Nice catch! I forgot to check that (which makes things a lot easier).